### ISSUE TITLE
Power interface and distribution

### ISSUE THREAD

**Issue by christiankuttke** [openUC2/openUC2_SOOP#3]
Design a power distribution solution that fits in the box, has 1 universal power input, and gives all devices their needed power.

**Comment by christiankuttke** [comment]
# Proposed input: 

220V AC Kaltgerätekabel (C14, Buchse C13) with fuse on the outside of the device. 

![image](https://github.com/openUC2/openUC2_SOOP/assets/74929609/28baa201-84d7-4f5a-a151-6541cf75a84a)

Power switch should have a power indicator light but should be protected from accidental turn off. 

C13 is available in a locking variant which is preferable, if available.

**Comment by christiankuttke** [comment]
# Devices to power, ordered by Power Nets

Estimates are worst-case and over-specced.

## 12 V

- Our uc32_standalone board featuring TMC2202 drivers, ESP32, interfaces for illumination, encoders, limit switches, PWM output (3A or more if necessary)
- Hard drives (HDD spinning) at spin-up 2A per drive

## 5V

- Raspberry Pi Screen (500 mA worst case)
- Nvidia Jetson Nano (3A)
- Raspberry Pi (3A)
- GPS module: 1A
- HDD: 2 A
- SSD: 1 A 

## Totals

worst-case, with 1 spinning HDD

12V 5A and 5V 6.5 A

All other devices can be powered by the boards and are probably covered by the generous estimates of those board's consumption.

**Comment by beniroquai** [comment]
This would be a high-level wiring chart:

<img width="1400" alt="image" src="https://github.com/openUC2/openUC2_SOOP/assets/4345528/a16261f9-3b45-44fa-aa1c-00d0f768520d">

now: 

`minimize(number(cables))`

**Comment by beniroquai** [comment]
File available here:
https://github.com/openUC2/openUC2_SOOP/blob/master/IMAGES/UC2_SOOP_Overview.pptx

In best case we have one single wire in. Let it be 12V to supply also the
SBC

**Comment by christiankuttke** [comment]
I would put a C13 jack on the outside of our case and power supplies
inside. The Kaltgerätekabel is universally available. Power supplies 220V
AC to 12 or 5V DC are available as THT component (Meanwell) to save space
in the box.

**Comment by beniroquai** [comment]
Wait, you want 230V inside the box? Absolutely not. This is difficult for many reasons. Let's keep it 12V only. I would recommend to use something like a through-hole connector for the 12V, no?
![image](https://github.com/openUC2/openUC2_SOOP/assets/4345528/3ecd1db3-aa58-41ea-b1ae-96dd5935e09f)

https://www.amazon.de/RUNCCI-YUN-Adapter-Hohlstecker%EF%BC%88Mit-Stecker-%C3%9Cberwachungskameras/dp/B089DXQDS9
this would directly go to the UC2 Standalone Board

C13 is this ?
![image](https://github.com/openUC2/openUC2_SOOP/assets/4345528/1f6bc3b6-d901-4288-a9cd-f06e8d091840)

**Comment by christiankuttke** [comment]
We need about 100W. Meanwell has neat desktop power supplies (bricks) with 90 or 120W output. But I looked around what the stock is and I found: For the 120W model, the round plug version (Meanwell product number ends with P1M) is only available at higher voltage. 

Most stock is of this model with 24V : GST120A24-P1M https://www.digikey.de/de/products/detail/mean-well-usa-inc/GST120A24-P1M/7703595

Second is this model with 20V (comparable to laptop power supplies): GST120A20-P1M https://www.digikey.de/de/products/detail/mean-well-usa-inc/GST120A20-P1M/7703593

Which means that the round plug is probably not suitable to carry above 6 amps.

The connector used for 12V 120W output is power DIN 4-pin locking a.k.a. KYCON KPPX-4P with per-pin capacity of 7.5A but run as a pair of 2 for each 12V and GND. The "shield" of the power DIN connector also passes through "Frame Ground", so probably Protective Earth of the power input. I assume we could connect metal parts of our system to that to "Ground" them for noise mitigation/ESD protection.

![image](https://github.com/openUC2/openUC2_SOOP/assets/74929609/2de5dd9f-1a88-4328-b223-55da93062205)
from datasheet of Meanwell GSM120A12-R7B

With that plug, there is Meanwell GSM series (medical vs. GST industrial) and there is this 120W 12V product with OK stock numbers: GSM120A12-R7B https://www.digikey.de/de/products/detail/mean-well-usa-inc/GSM120A12-R7B/7703447

The connector "locks" by virtue of being difficult to pull apart. I think that would be an OK connector on the outside of our box.

**Comment by beniroquai** [comment]
Why 100w?
The pi and display would not have more than 20 and the motor is not running
constantly. Maybe 12-20w max.

**Comment by christiankuttke** [comment]
To answer how I came to this value, please see the list above. 

To go from here, we could either spend 45€ on the 100W power supply and have that part of the box overengineered and overspecced, or we could spend 20€ on a no-name power supply with less wattage and then deal with errors or data loss in the case that everything does run at one at one unfortunate point in time and pull over the designed amount of power. 

The 100W power supply would eliminate power as a error source and allows us to build in quick-fixes like fans or other hard drive types/more drives without having to worry about power again at that point.

**Comment by beniroquai** [comment]
I would go for meanwell. I'm just scared about the 100W heat dissipation we may face in the box :D 
Why should we go for DIN vs the ordinary round 12V jack we have in the UC2 board? I remember you mentioned that but I didn't get it ;)

**Comment by beniroquai** [comment]
Sounds great. Let's go for the GST90A12-P1M. 
Power management (i.e. 12V -> 5V for the Raspi) is handled inside the box I guess, correct? Will there be an additional DC-DC converter like the one I gave you? Should be an off-the-shelf part I guess?

**Comment by beniroquai** [comment]
Hey @christiankuttke Now we have on12V input (Meanwell 5A@12V) that gets converted to 5A for the raspi (2x 3A@5V) DC-DC converter. Are there alternatives to e.g. provide 12V and 5V within one power supply? Would that make sense? Still, we are not getting all the power we need. Raspi is complaining and SSD is not running

**Comment by beniroquai** [comment]
Agree! Let's measure that on Friday. I ordered another 10A 5V DCDC converter to get more juice for the Pi. 

On the USB3 side.. (the pi has 2 of them, blue). Does it have a 600mA limit, still? Was there a software setting to get more power on the lines?

**Comment by beniroquai** [comment]
So from the DC DC converter it would be smarter to have a USBC adapter
powering the raspi? I tried that but didn't work an better.

**Comment by christiankuttke** [comment]
Re AC-DC power supply:

We used a Meanwell OWA-60E-12 power supply in the end. It has fixed Schuko and standard ID 2.1 x OD 5.5 x len 11 barrel jack with center positive. It has 5A output at 12V rated for 60W, and ramps voltage down to limit current when too much current is drawn (CC mode).

Supply: https://www.voelkner.de/products/902519/Mean-Well-OWA-60E-12-Tischnetzteil-Festspannung-12-V-DC-5A-60W.html

Datasheet: https://asset.re-in.de/add/160267/c1/-/en/001439263DS01/DA_Mean-Well-OWA-60E-12-Tischnetzteil-Festspannung-12-V-DC-5A-60W.pdf
